src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/_params.py

```diff
@@ -0,0 +1,139 @@
+register_cli_argument('sf', 'location',
+                      validator=get_default_location_from_resource_group)
+
+register_cli_argument('sf', 'secret_identifier',options_list=('--secret-identifier'),
+                      help='The existing Azure key vault secret URL')
+register_cli_argument('sf', 'certificate_file',options_list=('--certificate-file'),
+                     help='The existing certificate file path for the primary cluster certificate.')
+register_cli_argument('sf', 'parameter_file',options_list=('--parameter-file'),
+                     help='The path to the template parameter file.')
+register_cli_argument('sf', 'template_file',options_list=('--template-file'),
+                     help='The path to the template file.')
+register_cli_argument('sf', 'vm_password',options_list=('--vm-password'),
+                      help='The password of the Vm')
+register_cli_argument('sf', 'certificate_output_folder',options_list=('--certificate-output-folder'),
+                      help='The folder of the new certificate file to be created.')
+register_cli_argument('sf', 'certificate_password',options_list=('--certificate-password'),
+                      help='The password of the certificate file.')
+register_cli_argument('sf', 'certificate_subject_name',options_list=('--certificate-subject-name'),
+                      help='The subject name of the certificate to be created.')
+register_cli_argument('sf', 'vault_resource_group_name',options_list=('--vault-resource-group'),
+                      help='Azure key vault resource group name, it not given it will be defaulted to cluster resource group name')
+register_cli_argument('sf', 'vault_name',options_list=('--vault-name'),
+                      help='Azure key vault name, it not given it will be defaulted to the cluster resource group name')
+register_cli_argument('sf', 'cluster_size',options_list=('--cluster-size','-s'),
+                      help='The number of nodes in the cluster. Default are 5 nodes')
+register_cli_argument('sf', 'vm_sku', options_list=('--vm-sku'), help='The Vm Sku')
+register_cli_argument('sf', 'vm_user_name', options_list=('--vm-user-name'), help='The user name for logging to Vm. Default will be adminuser')
+register_cli_argument('sf', 'vm_os', default='WindowsServer2016Datacenter', options_list=('--vm-os', '-os'),
+                      help='The Operating System of the VMs that make up the cluster.',
+                     **enum_choice_list(['WindowsServer2012R2Datacenter', 'WindowsServer2016Datacenter', 'WindowsServer2016DatacenterwithContainers', 'UbuntuServer1604']))
+
+register_cli_argument('sf client certificate', 'certificate_common_name',
+                      help='Specify client certificate common name.')
+register_cli_argument('sf client certificate', 'admin_client_thumbprints',
+                      help='Specify list of client certificate thumbprint that only has admin permission.')
```

Comment:
How do you specify the list?  The only valid lists in the CLI are space-separated lists, and the help should call this out.